### DISCUSSION TITLE
Best way to handle pickup items

### DISCUSSION THREAD

**Discussion by SSNTails** [jrouwe/JoltPhysics#995] [Q&A] [answered]
What's the best way to handle interactions with collectible items, or items that the player interacts with that you don't want a dynamic collision response with? Think like coins or a fireflower in Mario, or weapons/ammo in Doom. Is OnContactValidate the place to record an interaction to process after the simulation completes, or is there a better way?

**Comment by MrOnlineCoder** [comment]
I have not yet reached point of implementing collectibles, but I guess you have just few options, although it will also depend on the amount of items you want to handle (tens or hunderds):
1) Handle collisions manually, without using Jolt - which I consider to be kinda stupid
2) Add a static box shape for each item and handle collisions as you described - via contact listeners. You may need to adjust the collision layers so that other objects like bullets or enemies will pass-through the item.
3) Use broadphase/narrowphase raycast in direction of player for given small distance to find out if player reached the item collection range.

Option 2 seems as most optimal for me, I guess handling even hundreds of such bodies won't be a problem for Jolt, assuming the fact they would be static.

**Reply by SSNTails** [reply]
Hmm, I wonder what the best way would be to keep track of those collisions in a list during OnContactValidate, without causing race conditions?

**Reply by jrouwe** [reply, marked as answer]
Or you place a 'sensor' around your pickup, see: https://github.com/jrouwe/JoltPhysics/blob/master/Samples/Tests/General/SensorTest.cpp
